internal/api/handler.go

```diff
@@ -1,27 +1,68 @@
 package api
 
 import (
+	"encoding/json"
 	"github.com/gin-gonic/gin"
 	"io"
 
 	"net/http"
 	"strings"
 )
 
+type StructEntrance struct {
+	PerformanceURL string `json:"url"`
+}
+
+type StructRes struct {
+	PerformanceResult string `json:"result"`
+}
+
 func (s *RestAPI) ShortenURLHandler(c *gin.Context) {
 	body, err := io.ReadAll(c.Request.Body)
 	if err != nil {
 		c.String(http.StatusInternalServerError, "Failed to read request body", http.StatusInternalServerError)
 		return
 	}
 	URLtoBody := strings.TrimSpace(string(body))
-
-	shortURL := s.StructService.GetShortURL(URLtoBody)
-
+	shortURL, err := s.StructService.GetShortURL(URLtoBody)
+	if err != nil {
+		c.String(http.StatusInternalServerError, "failed to record event to file", http.StatusInternalServerError)
+		return
+	}
 	c.Header("Content-Type", "text/plain")
 	c.String(http.StatusCreated, shortURL)
 }
 
+func (s *RestAPI) ShortenURLHandlerJSON(c *gin.Context) {
+	body, err := io.ReadAll(c.Request.Body)
+	if err != nil {
+		c.String(http.StatusInternalServerError, "Failed to read request body", http.StatusInternalServerError)
+		return
+	}
+
+	var decoderBody StructEntrance
+	err = json.Unmarshal(body, &decoderBody)
+	if err != nil {
+		c.String(http.StatusInternalServerError, "Failed to read request body", http.StatusInternalServerError)
+		return
+	}
+	URLtoBody := strings.TrimSpace(decoderBody.PerformanceURL)
```

Comment:
```suggestion
	url := strings.TrimSpace(decoderBody.PerformanceURL)
```
URLtoBody это действие, ожидается что там анонимная функция, но никак не строка